x-pack/plugins/observability_solution/inventory/public/components/search_bar/index.tsx

```diff
@@ -0,0 +1,91 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+import { SearchBarOwnProps } from '@kbn/unified-search-plugin/public/search_bar';
+import deepEqual from 'fast-deep-equal';
+import React, { useCallback, useEffect } from 'react';
+import { Subject } from 'rxjs';
+import { i18n } from '@kbn/i18n';
+import { EntityType } from '../../../common/entities';
+import { useAdHocInventoryDataView } from '../../hooks/use_adhoc_inventory_data_view';
+import { useInventoryParams } from '../../hooks/use_inventory_params';
+import { useKibana } from '../../hooks/use_kibana';
+import { EntityTypesControls } from './entity_types_controls';
+
+export const searchBarContentSubject$ = new Subject<{
+  kuery?: string;
+  entityTypes?: EntityType[];
+  refresh: boolean;
+}>();
```

Comment:
can you be more specific, please?

Reply:
The way this is declared means that it's a globally accessible subject. Wrapping it in a context keeps the subject scoped within the component tree. I'm not sure, but it's possible that importing this in multiple files will create X instances of the subject. 

Besides, it would make it easier to test the components and mock the subject.